modules/gapi/include/opencv2/gapi/cpu/gcpukernel.hpp

```diff
@@ -304,17 +338,16 @@ struct OCVCallHelper<Impl, std::tuple<Ins...>, std::tuple<Outs...> >
         //them to parameters of ad-hoc function
         //Convert own::Scalar to cv::Scalar before call kernel and run kernel
         //convert cv::Scalar to own::Scalar after call kernel and write back results
-        call_and_postprocess<decltype(get_in<Ins>::get(ctx, IIs))...>
-                                      ::call(get_in<Ins>::get(ctx, IIs)...,
-                                             get_out<Outs>::get(ctx, OIs)...);
+        Impl::run(get_in<Ins>::get(ctx, IIs)...,
+                  get_out<Outs>::get(ctx, OIs)...);
+        postprocess(get_out<Outs>::get(ctx, OIs)...);
     }
 
     template<int... IIs, int... OIs>
     static void call_impl(cv::GCPUContext &ctx, Impl& impl, detail::Seq<IIs...>, detail::Seq<OIs...>)
     {
-        call_and_postprocess<decltype(cv::detail::get_in<Ins>::get(ctx, IIs))...>
-                                      ::call(impl, cv::detail::get_in<Ins>::get(ctx, IIs)...,
-                                                   cv::detail::get_out<Outs>::get(ctx, OIs)...);
+        impl(cv::detail::get_in<Ins>::get(ctx, IIs)...,
+             cv::detail::get_out<Outs>::get(ctx, OIs)...);
```

Comment:
one-liner

Reply:
Also please put `postprocess` here. I wrote already about that old bug.

Reply:
okay, no problem, I thought about another Anatoliy's feature - OSD Render and not Kernel as lambda